insights/parsers/subscription_manager_list.py

```diff
@@ -0,0 +1,231 @@
+        """
+        Search for records that match the given keys and values.  See the
+        :func:`insights.parsers.keyword_search` function for more details
+        on usage.
+
+        Arguments:
+            **kwargs: Key-value pairs of search parameters.
+
+        Returns:
+            (list): A list of records that matched the search criteria.
+
+        >>> consumed = shared[SubscriptionManagerListConsumed]
+        >>> len(consumed.search(Service_Level='PREMIUM'))
+        1
+        >>> consumed.search(Provides__contains='Red Hat Enterprise Virtualization')
+        []
+        """
+        return keyword_search(self.records, **kwargs)
+
+
+@parser('subscription_manager_list_consumed')
+class SubscriptionManagerListConsumed(SubscriptionManagerList):
+    """
+    Read the output of ``subscription-manager list --consumed``.
+
+    Sample input file::
+
+        +-------------------------------------------+
+           Consumed Subscriptions
+        +-------------------------------------------+
+        Subscription Name: Red Hat Enterprise Linux Server, Premium (1-2 sockets) (Up to 1 guest)
+        Provides:          Oracle Java (for RHEL Server)
+                           Red Hat Software Collections Beta (for RHEL Server)
+                           Red Hat Enterprise Linux Server
+                           Red Hat Beta
+        SKU:               RH0155783S
+        Contract:          12345678
+        Account:           1000001
+        Serial:            0102030405060708090
+        Pool ID:           8a85f981477e5284014783abaf5d4dcd
+        Active:            True
+        Quantity Used:     1
+        Service Level:     PREMIUM
+        Service Type:      L1-L3
+        Status Details:    Subscription is current
+        Subscription Type: Standard
+        Starts:            11/14/14
+        Ends:              07/06/15
+        System Type:       Physical
+
+    Examples:
+        >>> consumed = shared[SubscriptionManagerListConsumed]
+        >>> type(consumed)
+        <class 'insights.parsers.subscription_manager_list.SubscriptionManagerListConsumed'>
+        >>> len(consumed.records)
+        1
+        >>> sub1 = consumed.records[0]
+        >>> sub1['SKU']
+        'RH0155783S'
+        >>> sub1['Active']  # Type conversion on Active field
+        True
+        >>> sub1['Status Details']  # Keys appear as given
+        'Subscription is current'
+        >>> type(sub1['Provides'])  # Provides lines as a list
+        <type 'list'>
+        >>> sub1['Provides'][1]
+        'Red Hat Software Collections Beta (for RHEL Server)'
+        >>> sub1['Starts']  # Basic field as text - note month/day/year
+        '11/14/14'
+        >>> type(sub1['Starts timestamp'])  # Converted to date
+        <type 'datetime.date'>
+        >>> sub1['Starts timestamp'].year
+        2014
+        >>> consumed.all_current  # Are all subscriptions listed as current?
+        True
+    """
+    @property
+    def all_current(self):
```

Comment:
Just a suggestion too, Instead of calculating the "Boolean" result each time when calling this interface, how about just calculating it once and assign the result to an attribute to save some calling workload?

Reply:
I thought about that, but those properties are only relevant for the child, not the parent.  So to make it an attribute would mean overriding the `parse_content` method in the subclass and doing the post-processing in that.  To me that's a little less elegant.

I agree that it will have to rescan each time it's called, but a) hopefully that won't be too often, and b) there aren't usually a lot of records to scan.

The other benefit of doing it this way is that if someone never uses this, it's not called and doesn't waste any time pre-calculating.

Reply:
sounds good to me :)